apps/frontend/src/pages/Profile.tsx

```diff
@@ -0,0 +1,264 @@
+  const isScreenSmall = useMediaQuery(theme.breakpoints.down("lg"));
+  const { id } = useParams();
+
+  const [, setImage] = useState<File | null>(null);
+  const [imageURL, setImageURL] = useState<string>("");
+  const [profile, setProfile] = useState({
+    name: "",
+    username: "",
+    password: "",
+    email: "",
+    bio: "",
+    profile_picture: "",
+  });
+  const [originalProfile, setOriginalProfile] = useState(profile);
+  const [editMode, setEditMode] = useState(false);
+  const [isImageUploaded, setIsImageUploaded] = useState(false);
+
+  // Load the user info
+  useEffect(() => {
+    _axios_instance
+      .get("/user/" + id)
+      .then((response) => {
+        setProfile(response.data);
+        setOriginalProfile(response.data);
+      })
+      .catch((error) => {
+        console.log(error);
+      });
+  }, [id]);
+
+  // This function makes sure that the passed in url is a base64 data string
+  const isImageValid = (url: string) => {
+    try {
+      return url.startsWith("data:image/");
+    } catch (error) {
+      return false;
+    }
+  };
+
+  // Uploads a single image to the S3 server
+  const asyncUploadSingleImage = async (
+    imageBinary: string,
+  ): Promise<ImageURLObject | false> => {
+    try {
+      // Attempt to upload the image
+      const response = await _axios_instance.post("/images", imageBinary, {
+        headers: {
+          "Content-Type": "multipart/form-data",
+        },
+      });
+
+      // Return the URL of the uploaded image on success
+      return { url: response.data.url };
+    } catch (error) {
+      // Log error and return false on failure
+      console.error("Error uploading image:", error);
+      return false;
+    }
+  };
+
+  const handleImageUpload = (event: React.ChangeEvent<HTMLInputElement>) => {
+    if (event.target?.files && event.target?.files.length > 0) {
+      const file = event.target?.files[0];
+
+      if (file) {
+        const reader = new FileReader();
+
+        reader.onload = function (event) {
+          const base64String = event.target?.result as string;
+          setImageURL(base64String); // Base64 string of the file contents
+
+          if (!isImageValid(base64String)) {
+            alert("Invalid image type. Please upload a valid image file.");
+            return;
+          }
+
+          setImage(file);
+          setImageURL(URL.createObjectURL(file));
+          setEditMode(true);
+        };
+
+        reader.readAsDataURL(file); // This will trigger the onload function above
+      }
+    } else {
+      alert("No image selected.");
+    }
+
+    setIsImageUploaded(true);
+  };
+
+  const handleInputChange = (
+    e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>,
+    field: string,
+  ) => {
+    setProfile({ ...profile, [field]: e.target.value });
+    setEditMode(true);
+  };
+
+  const handleSaveChanges = async () => {
+    let successImages: boolean | ImageURLObject = false;
```

Comment:
It would be more idiomatic to use null instead of false to represent an absence of an image